dcontext/hardsoft.go

```diff
@@ -143,6 +143,10 @@ func HardContext(softCtx context.Context) context.Context {
 	if parentHardCtx == nil {
 		return softCtx
 	}
+	// If it's already a hard context, just return it as is.
+	if hardCtx, ok := softCtx.(childHardContext); ok {
+		return hardCtx
+	}
```

Comment:
I'm confused, because I'd have told you that the above `if parentHardCtx == nil` case would cover this.

Reply:
Ohh, I get it, it's the same type of bug as https://github.com/datawire/dlib/pull/26